playbooks/common/openshift-master/config.yml

```diff
@@ -235,36 +244,24 @@
   - role: haproxy
     when: groups.oo_masters_to_config | length > 1
 
-- name: Generate master session keys
+- name: Generate master session secrets
   hosts: oo_first_master
-  tasks:
-  - fail:
-      msg: "Both openshift_master_session_auth_secrets and openshift_master_session_encryption_secrets must be provided if either variable is set"
-    when: (openshift_master_session_auth_secrets is defined and openshift_master_session_encryption_secrets is not defined) or (openshift_master_session_encryption_secrets is defined and openshift_master_session_auth_secrets is not defined)
-  - fail:
-      msg: "openshift_master_session_auth_secrets and openshift_master_encryption_secrets must be equal length"
-    when: (openshift_master_session_auth_secrets is defined and openshift_master_session_encryption_secrets is defined) and (openshift_master_session_auth_secrets | length != openshift_master_session_encryption_secrets | length)
-  - name: Install OpenSSL package
-    action: "{{ansible_pkg_mgr}} pkg=openssl state=present"
-  - name: Generate session authentication key
-    command: /usr/bin/openssl rand -base64 24
-    register: session_auth_output
-    with_sequence: count=1
-    when: openshift_master_session_auth_secrets is undefined
-  - name: Generate session encryption key
-    command: /usr/bin/openssl rand -base64 24
-    register: session_encryption_output
-    with_sequence: count=1
-    when: openshift_master_session_encryption_secrets is undefined
+  vars:
+    session_secrets_missing: "{{ not (openshift.master.session_auth_secrets | default(false) or openshift.master.session_encryption_secrets | default(false)) }}"
+  pre_tasks:
   - set_fact:
-      session_auth_secret: "{{ openshift_master_session_auth_secrets
-                                | default(session_auth_output.results
-                                | oo_collect(attribute='stdout')
-                                | list) }}"
-      session_encryption_secret: "{{ openshift_master_session_encryption_secrets
-                                      | default(session_encryption_output.results
-                                      | oo_collect(attribute='stdout')
-                                      | list) }}"
+      session_auth_secrets: "{{ [ 24 | oo_generate_secret ] }}"
+      session_encryption_secrets: "{{ [ 24 | oo_generate_secret ] }}"
```

Comment:
Nevermind...  something like this would probably work better:
```
  pre_tasks:
  - set_fact:
      openshift_session_auth_secrets: "{{ openshift.master.session_auth_secrets | default([24 | oo_generate_secret]) }}"
    when openshift_session_auth_secrets is not defined
  - set_fact:
      openshift_session_encryption_secrets: "{{ openshift.master.session_encryption_secrets | default([24 | oo_generate_secret]) }}"
    when openshift_session_encryption_secrets is not defined
```